api/method_edges.go

```diff
@@ -50,6 +50,52 @@ func (s *Server) listEdgesHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (s *Server) listMiniStakesHandler(w http.ResponseWriter, r *http.Request) {
+	specEdges, err := s.edges.GetEdges(r.Context())
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, err)
+		return
+	}
+	e, err := convertSpecEdgeEdgesToEdges(r.Context(), specEdges)
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, err)
+		return
+	}
+	stakeInfoMap := make(map[common.Hash]map[string]*StakeInfo)
+	for _, edge := range e {
+		if (edge.MiniStaker != common.Address{}) {
+			if stakeInfoMap[edge.AssertionHash] == nil {
+				stakeInfoMap[edge.AssertionHash] = make(map[string]*StakeInfo)
+			}
+			if stakeInfoMap[edge.AssertionHash][edge.Type] == nil {
+				stakeInfoMap[edge.AssertionHash][edge.Type] = &StakeInfo{
+					StakerAddresses:       []common.Address{},
+					TotalMinistakes:       0,
+					StartCommitmentHeight: edge.StartCommitment.Height,
+					EndCommitmentHeight:   edge.EndCommitment.Height,
+				}
+			}
+			stakeInfoMap[edge.AssertionHash][edge.Type].StakerAddresses = append(stakeInfoMap[edge.AssertionHash][edge.Type].StakerAddresses, edge.MiniStaker)
+			stakeInfoMap[edge.AssertionHash][edge.Type].TotalMinistakes++
```

Comment:
Completely agree on this, but currently we only have a fixed stake amount (the implementation for variable staking is not in place at the moment i think).
Since we have a fixed stake right now, adding that information does not seems that helpful (also since the stake amount is constant, its not attached to the edge proto, so fetching it is non trivial, but with the variable staking implementation that information will be attached to the edge and will be easier to fetch), will update the code once the variable staking is in place.

Also changing TotalMinistakes to NumberOfMinistakes to avoid confusion.

Reply:
Agreed here that number of ministakes is helpful because it helps us gauge if there is a sybil more clearly